providers.tf.example

```diff
@@ -14,16 +14,22 @@ provider "aws" {
 provider "kubernetes" {
   host                   = data.aws_eks_cluster.cluster.endpoint
   cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority[0].data)
-  token                  = data.aws_eks_cluster_auth.cluster.token
+
+  # https://registry.terraform.io/providers/hashicorp/kubernetes/latest/docs#exec-plugins
+  exec {
+    api_version = "client.authentication.k8s.io/v1beta1"
+    args        = ["eks", "get-token", "--profile", "change_me", "--cluster-name", data.aws_eks_cluster.cluster.name]
```

Comment:
Is it possible to set profile one time in file and then reuse it on all places in file as variable?

Reply:
I don't know how to set file local variables in Terraform. Wouldn't it be better to have an entry in `locals.tf` or `variables.tf`? At least in our local _Infrastructure_ repository we have the profile value as entry in our `variables.tf`.

Reply:
But since this is just an example, there is no real need to have it as customizable variable here!

Reply:
Since ```profile``` is also used   in ```state-backend-template```, adding ```profile``` and even ```region``` to the ```variables.tf``` might be a good idea.

Reply:
Sorry for incorrect previous comment, backend block cannot refer named values.
But still, having ```profile``` and ```region``` in ```variables.tf``` seems correct.

Reply:
We will refactor profile setup and usage in different task/PR.